shell/browser/electron_browser_client.cc

```diff
@@ -1327,6 +1338,21 @@ void ElectronBrowserClient::CreateWebSocket(
   v8::Isolate* isolate = v8::Isolate::GetCurrent();
   v8::HandleScope scope(isolate);
   auto* browser_context = frame->GetProcess()->GetBrowserContext();
+
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+  auto* web_request_api =
+      extensions::BrowserContextKeyedAPIFactory<extensions::WebRequestAPI>::Get(
+          browser_context);
+
+  DCHECK(web_request_api);
+
+  if (web_request_api->MayHaveProxies()) {
+    web_request_api->ProxyWebSocket(frame, std::move(factory), url,
```

Comment:
I think this needs the same treatment as `WillCreateURLLoaderFactory`? i.e. we need to check the return value here and early-exit if `extensions::WebRequestAPI` decided to take over the request.